weave/flow/dataset.py

```diff
@@ -71,6 +74,50 @@ def to_pandas(self) -> "pd.DataFrame":
 
         return pd.DataFrame(self.rows)
 
+    def add_rows(self, rows: Iterable[dict]) -> "Dataset":
+        """Create a new dataset version by appending rows to the existing dataset.
+
+        This is useful for adding examples to large datasets without having to
+        load the entire dataset into memory.
+
+        Args:
+            rows: The rows to add to the dataset.
+
+        Returns:
+            The updated dataset.
+        """
+        client = get_weave_client()
+        if client is None:
+            raise ValueError("Weave is not initialized, call weave.init().")
+        if not isinstance(self.rows, WeaveTable) or not self.rows.table_ref:
+            raise TypeError(
+                "This dataset is not saved to weave. Call weave.publish(dataset) "
+                "to save the dataset before adding rows."
+            )
+        if self.rows.table_ref.project != client.project:
+            raise ValueError(
+                "This dataset is not saved to the same project as the current weave client. "
+                "Client is in project:  "
+                + client.project
+                + " but dataset is in project: "
+                + self.rows.table_ref.project
+            )
```

Comment:
Why does this matter?

Reply:
We cannot create a dataset that points to a table in a different project